flagsmith-engine/segments/evaluators.ts

```diff
@@ -55,18 +55,21 @@ function traitsMatchSegmentRule(
     );
 }
 
-function traitsMatchSegmentCondition(
+export function traitsMatchSegmentCondition(
     identityTraits: TraitModel[],
     condition: SegmentConditionModel,
     segmentId: number | string,
     identityId: number | string
 ): boolean {
-    if (condition.operator == PERCENTAGE_SPLIT) {
-        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(condition.value);
-    }
-
     const traits = identityTraits.filter(t => t.traitKey === condition.property_);
     const trait = traits.length > 0 ? traits[0] : undefined;
-
-    return trait ? condition.matchesTraitValue(trait.traitValue) : false;
-}
+    if (condition.operator == PERCENTAGE_SPLIT) {
+        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(<string>condition.value);
+    } else if (condition.operator === IS_SET ) {
+        return traits.length > 0
```

Comment:
We have a couple of options to improve this: 

 1. Since, at the point, we've already evaluated the length of the traits array, we shouldn't do that again. We should just use `!!trait`
 2. Alternatively, since we don't actually need the `trait` at this point, we could move that line below the logic to evaluate the `IS_SET` / `IS_NOT_SET` operators.